src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -170,134 +167,204 @@ public int updateInternal(Position p) {
-                + "      t.\"positionUuid\" = \"peoplePositions\".\"positionUuid\" AND"
-                + "      t.\"createdAt\" = \"peoplePositions\".\"createdAt\" AND"
-                + "      \"peoplePositions\".\"endedAt\" IS NULL";
-      } else {
-        sql =
-            "/* positionSetPerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt FROM "
-                + "(SELECT * FROM \"peoplePositions\""
-                + " WHERE \"personUuid\" = :personUuid AND \"positionUuid\" = :positionUuid AND \"endedAt\" IS NULL"
-                + " ORDER BY \"createdAt\" DESC LIMIT 1) AS t "
-                + "WHERE t.\"personUuid\" = \"peoplePositions\".\"personUuid\" AND"
-                + "      t.\"positionUuid\" = \"peoplePositions\".\"positionUuid\" AND"
-                + "      t.\"createdAt\" = \"peoplePositions\".\"createdAt\" AND"
-                + "      \"peoplePositions\".\"endedAt\" IS NULL";
-      }
-      getDbHandle().createUpdate(sql).bind("personUuid", personUuid)
-          .bind("positionUuid", currPos.getUuid()).bind("endedAt", DaoUtils.asLocalDateTime(now))
-          .execute();
+    // If the person is already assigned to another position, remove the position from the person
+    numRows += AnetObjectEngine.getInstance().getPositionDao().removePositionFromPerson(personUuid);
 
-      getDbHandle()
-          .createUpdate(
-              "/* positionSetPerson.remove1 */ UPDATE positions set \"currentPersonUuid\" = NULL "
-                  + "WHERE \"currentPersonUuid\" = :personUuid")
-          .bind("personUuid", personUuid).execute();
 
-      getDbHandle()
-          .createUpdate("/* positionSetPerson.remove2 */ INSERT INTO \"peoplePositions\" "
-              + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
-              + "VALUES (:positionUuid, NULL, :createdAt)")
-          .bind("positionUuid", currPos.getUuid()).bind("createdAt", DaoUtils.asLocalDateTime(now))
-          .execute();
-    }
+    // Get timestamp *after* remove to preserve correct order
+    final Instant now = Instant.now();
 
-    getDbHandle()
+    numRows += getDbHandle()
         .createUpdate("/* positionSetPerson.set1 */ UPDATE positions "
             + "SET \"currentPersonUuid\" = :personUuid WHERE uuid = :positionUuid")
         .bind("personUuid", personUuid).bind("positionUuid", positionUuid).execute();
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + " WHERE \"positionUuid\" = :positionUuid AND " + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid)
+        .bind("endedAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
```

Comment:
Please ensure that each database statement has a unique comment.

Reply:
Fixed with using unique comments.